clippy_lints/src/attrs/mod.rs

```diff
@@ -526,6 +527,47 @@ declare_clippy_lint! {
     "duplicated attribute"
 }
 
+declare_clippy_lint! {
+    /// ### What it does
+    /// Checks for items annotated with `#[forbid(...)]`.
+    ///
+    /// ### Why is this bad?
+    /// Banning certain coding practices outright might be harmful,
+    /// since there might very well be reasonable exceptions.
+    ///
+    /// ### Example
+    /// ```no_run
+    /// #![forbid(panic)]
+    /// ```
+    ///
+    /// Use instead:
+    /// ```no_run
+    /// #![deny(panic)]
+    /// ```
+    #[clippy::version = "1.79.0"]
+    pub FORBID,
+    restriction,
+    "usage of `forbid(...)`"
+}
+
+declare_clippy_lint! {
+    /// ### What it does
+    /// Checks for items annotated with `#[forbid(clippy::forbid)]`.
+    ///
+    /// ### Why is this bad?
+    /// Prohibiting others from banning anything is hypocritical.
+    /// Additionally, there might be valid reasons to ban certain coding practices in certain projects.
+    ///
+    /// ### Example
+    /// ```no_run
+    /// #![forbid(clippy::forbid)]
+    /// ```
+    #[clippy::version = "1.79.0"]
+    pub HYPOCRISY,
+    suspicious,
+    "usage of `forbid(clippy::forbid)`"
+}
+
```

Comment:
I was wondering if `#![forbid(clippy::forbid)]` will trigger the lint. If it does, the hypocrisy lint seems redundant.

Reply:
There is no good answer for that, it should depend on whether it's inner or outer I think, as such I think it should intentionally not lint itself since this would be confusingly inconsistent

I don't really think this extra lint is necessary to have. If somebody wants to forbid use of forbid, then that's ok even despite the listed arguments imo. The argument of false positives doesn't apply here since ig it's just... too simple.

Also, bikeshedding, naming it `hypocrisy` makes it sound more general than it is. It only applies to `forbid`.